src/client.rs

```diff
@@ -62,11 +63,64 @@ impl CoinGeckoClient {
     /// let client = CoinGeckoClient::new("https://some.url");
     /// ```
     pub fn new(host: &'static str) -> Self {
-        CoinGeckoClient { host }
+        CoinGeckoClient {
+            host,
+            api_key: None,
+        }
+    }
+
+    /// Creates a new CoinGeckoClient client with a custom host url and API key
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use coingecko::CoinGeckoClient;
+    /// let client = CoinGeckoClient::new_with_api_key("https://some.url", "123456789");
+    /// ```
+    pub fn new_with_api_key(host: &'static str, api_key: &'static str) -> Self {
+        CoinGeckoClient {
+            host,
+            api_key: Some(api_key),
+        }
     }
 
-    async fn get<R: DeserializeOwned>(&self, endpoint: &str) -> Result<R, Error> {
-        reqwest::get(format!("{host}/{ep}", host = self.host, ep = endpoint))
+    /// Gets a URL for the provided endpoint and optional params.
+    /// The endpoint must be prefixed with a /.
+    /// If params are specified, they must be prefixed with a ?.
+    /// If an API key is present, the x_cg_pro_api_key param is added.
```

Comment:
Thanks for adding these docs, seems like an oversight for them to have left that out before. Can you change it to this format though...
```suggestion
    /// Gets a URL for the provided endpoint and optional params.
    /// 
    /// **Note:** If an API key is present, the `x_cg_pro_api_key` param is added automatically.
    /// this will be appended to `params` if necessary.
    /// 
    /// - `endpoint`: API endpoint, must be prefixed with a /. e.g. `"/simple/price"`.
    /// - `params`: API endpoint parameters to append, they must be prefixed with a `'?'`. e.g. `"?ids=1"`.
```